### ISSUE TITLE
Ajax auto load only works the first time after page load?

### ISSUE THREAD

**Issue by jjnxpct** [Joomline/JlContentFieldsFilter#57]
Hi! We have implemented jcontenfieldsfilter on one of our websites. We have created 4 custom fields of the list type and use these to filter a specific article category on the website. This works fine, except for the ajax functionality. When we enable ajax and the auto submit feature this only works once after the page loads. When we change the filter after the first time nothing happens. 

I don't see any errors in the console.

We are on Joomla! 3.9.20
PHP 7.3.19
JL Content Fields Filter 1.8.2

Any thought?

**Comment by AlekVolsk** [comment]
You need to bring the filter module outside the updated content block

**Comment by jjnxpct** [comment]
I inserted the module into the component view (override we created) to be able to place it under the page title. But I guess that creates this ajax issue. I will try to do this another way then. Thanks for pointing me in the right direction!

**Comment by jjnxpct** [comment]
One more thing about this if I may: It might be helpful (to me at least) to be able to target the reload div where the items of the blog view are loaded. Not the whole page. This way I could leave the module where it is, and then let the Ajax refresh on;y the list items on the page. (For example all div.item). I could wrap those in a div and set that as a target in the module. But when I do this now the whole page gets loaded in that div. But this might not be that easy... It;s just a thought.

**Comment by zikkuratvk** [comment]
see
https://joomline.org/docs/77-jl-content-fields-filter/407-installation-and-setting-jl-content-fields-filter.html

> Setting «Ajax container selector» is needed when choosing an ajax filtering. Here you need to enter the unique ID of the main container that displays the content. The ID is written according to the rules of writing CSS selectors.

**Comment by jjnxpct** [comment]
Yes, I have seen that. But when I select the containing page div it includes the module because we have loaded the module in the page view... If we only select the div that contains the listed items on the page it will replace this div with the whole page, including the module itself. Or am I missing something here?

Anyway, we are creating these issues ourself by loading the module in the page view. So maybe we need to place the module somewhere else outside the page (component) view. Than we won;t have this issue and we can use the Ajax as it is supposed to work.

But I paced the filter on top of  the list (below the title and an intro text) because I think that's a better way to present it. If we place the module above the title and intro it's not as user friendly. So this is why we placed it inside the component view. 

My suggestion was that we could use just the containing div of the items on the page to reload those with ajax. Instead of reloading the whole component view. But this might also raise some other issues like updating the pagination... 

So I guess I have to choices:
1. Place the module outside the component view and then we can use the ajax functionality
2. Leave the module inside the component view but then not use the ajax functionality.

**Comment by jjnxpct** [comment]
If have chosen option 1 ;-) I do like the ajax functionality so now it works like it should. I hab placed the module in the template index  before the component output. I had to do some other shuffling around but it now works as I want. Thanks for you patience...

**Comment by AlekVolsk** [comment]
Thank you for using our solution and we were useful to you)
